protocol/src/protocol.progress.proposed.md

```diff
@@ -0,0 +1,63 @@
+#### Reporting server task progress
```

Comment:
Before we merge this in I would like that we at least think about how cancellation would work in this model since I think that this is a common pattern with progress. Options I see:

- we have a notification from the client to a server to signal cancellation based on the progress id
- we make the progress notification a request that returns whether this is canceled or not (not my favorite)
- we allow the progress to refer to a open request so that a client can map the things together. 
- we support that requests can provide a progress id which the server then uses to report progress on (my preferred options).

Reply:
Things to consider:
- The request ID is usually not known to the implementation of a method, it's hidden in the RPC layer.
- A middleware proxy might rewrite request IDs (e.g. prefix them if managing many servers) but wouldn't be aware of also having to rewrite request IDs in the params of the progress notification.
- Progresses can often not correspond to requests, but to async actions like indexing and background initialisation, dependency installation, ... These may sometimes be cancellable (although usually you'd really want to just shut down the server), but cannot be cancelled by a request ID.

Option 1 would be a possible way forward in the future that considers these (although this proposal should not be blocked on that).

Reply:
Ditto what @felixfbecker said - I wouldn't want to block this and also it's worth thinking that progress report isn't always done against a client-issued request and doesn't have to support cancellation in every case.